### PR TITLE
fix(reports): monthly analysis of balance cleanup

### LINKED ISSUE #5242: Analyze mensuel de la balance bugs
Our report "analyze mensuel de la balance" has two bugs in it:

  1. Selecting a single account doesn't work.
  2. The report doesn't use the enterprise currency (it's always $).

### PR DESCRIPTION
**Pull request by jniles**
This commit cleans up several items in the monthly analysis of the
balance sheet report.

1. The report is migrated to async/await.
2. Properly uses the tree library for calculation of totals.
3. Uses the `bhAccountSelect` to select accounts.
4. Makes the currency render in the enterprise currency.

Close #5242.

image

### CODE REVIEW

File: client/src/modules/reports/generate/monthly_balance/monthly_balance.html [hunk 1]
```diff
@@ -68,27 +68,11 @@ <h3 class="text-capitalize" translate>REPORT.MONTHLY_BALANCE.TITLE</h3>
...
-
-                <div class="help-block" ng-messages="ConfigForm.account_id.$error" ng-show="ConfigForm.$submitted">
-                  <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
-                </div>
-              </div>
+              <bh-account-select
+                account-id="ReportConfigCtrl.reportDetails.accountId"
+                on-select-callback="ReportConfigCtrl.onSelectAccount(account)"
+                exclude-title-accounts="true">
+              </bh-account-select>
```
Comment (mbayopanda) [inline client/src/modules/reports/generate/monthly_balance/monthly_balance.html:75]: 👍🏽

File: server/controllers/finance/reports/monthly_balance/index.js [hunk 2]
```diff
@@ -76,52 +68,49 @@ function reporting(opts, session) {
...
+    JOIN period AS p ON p.id = pt.period_id
+    WHERE pt.fiscal_year_id = ? AND pt.period_id = ?
+  `;
+
+  const [exploitation, totalExploitation] = await Promise.all([
+    db.exec(sql, sqlParams),
+    db.one(sqlSum, sqlParams),
+  ]);
+
+  const rows = exploitation.length === 0 ? [] : prepareTree(exploitation, 'amount', 'debit', 'credit');
```
Comment (mbayopanda) [inline server/controllers/finance/reports/monthly_balance/index.js:89]: 👍🏽

**Review by mbayopanda — COMMENTED** [review]
The code is more clean and clear but I got this error on both the test and a production database

image

**Comment by jniles** [comment]
Please check your sql-mode variable.  It's clearly not set correctly.  It's the same reason that your tests don't pass.   See https://github.com/IMA-WorldHealth/bhima/issues/5842

**Review by mbayopanda — APPROVED** [review]
LGTM 👍🏽 

bors r+

**Pull request merged**